### ISSUE TITLE
PDF Prints Blank on MacOS

### ISSUE THREAD

**Issue by ifarwell** [qzind/tray#422]
A QZ client encountered an issue on MacOS when printing a PDF through QZ -- the PDF is printing blank.  When printing with PDFBOX command line, the label prints as expected.  Unable to determine if this is a PDFBOX issue or a QZ issue.

This issue seems to be isolated to Mac - the label printed as expected with QZ on Windows 10.

**Environment:**
- Mac OS 10.13.6
- Zebra LP2844
- Label size: 57mm x 32mm (2.25in x 1.25in)

**(Quoting our Client)**
> To reproduce my problem:
> 1. Use the default Zebra LP2844 driver on a Mac.
> 2. Print any PDF from your demo page.
> 3. The label comes out blank.

PDF provided by client; permission was given to use this file for testing: [testPDF.pdf](https://github.com/qzind/tray/files/2908616/testPDF.pdf)

**Comment by ifarwell** [comment]
I tested the testPDF on Windows 10 with default Zebra driver for LP2844-Z (4x6 label), and the label printed as expected.

![testpdf_windows10_lp2844z_result](https://user-images.githubusercontent.com/44090703/53467040-4dcb9f00-3a22-11e9-94b7-7d1bca8d9823.jpg)

**Comment by bberenz** [comment]
Printing pdf's, even the sample one works fine on my mac. 
However, a huge caveat to this, is I do not have a printer using the zebra driver.

The client kindly linked their debug logs, which despite some binary issues, I extracted the following:
```
2019-02-14 17:47:54,053 [INFO] Connection opened from /fe80:0:0:0:0:0:0:1%1:51936 on socket port 8182
2019-02-14 17:47:54,127 [DEBUG] Message: {"certificate":<redacted>}
2019-02-14 17:47:54,132 [DEBUG] Received new certificate from connection through 51936
2019-02-14 17:47:54,347 [DEBUG] Message: {"call":"printers.find","promise":{},"params":{"query":"Zebra LP2844"},"signature":<redacted>,"timestamp":1550159274145,"uid":"e9dhaz","position":{"x":720,"y":421.5}}
2019-02-14 17:47:54,355 [DEBUG] Searching for PrintService matching Zebra LP2844
2019-02-14 17:47:54,355 [DEBUG] Found 4 printers
2019-02-14 17:47:54,442 [DEBUG] Found match: Zebra LP2844
2019-02-14 17:47:54,579 [DEBUG] Message: {"call":"print","promise":{},"params":{"printer":{"name":"Zebra LP2844"},"options":{"colorType":"grayscale","copies":1,"density":120,"duplex":false,"fallbackDensity":null,"interpolation":"nearest-neighbor","jobName":null,"legacy":false,"margins":0,"orientation":null,"paperThickness":null,"printerTray":null,"rasterize":false,"rotation":0,"scaleContent":"false","size":{"width":"5.71","height":"3.17"},"units":"cm","altPrinting":false,"encoding":null,"endOfDoc":null,"perSpool":1,"chunkSize":"50000"},"data":[{"type":"pdf","format":"file","data":<pdf>}]},"signature":<redacted>,"timestamp":1550159274479,"uid":"fyyb00","position":{"x":720,"y":421.5}}
2019-02-14 17:47:54,586 [DEBUG] Using qz.printer.action.PrintPDF to print
2019-02-14 17:47:54,586 [DEBUG] Searching for PrintService matching Zebra LP2844
2019-02-14 17:47:54,587 [DEBUG] Found 4 printers
2019-02-14 17:47:54,639 [DEBUG] Found match: Zebra LP2844
2019-02-14 17:47:54,640 [DEBUG] Found Resolution: 20300x20300 dphi
2019-02-14 17:47:54,862 [DEBUG] Parsed 1 files for printing
2019-02-14 17:47:54,909 [INFO] Adjusting orientation to print landscape PDF source
2019-02-14 17:47:54,909 [INFO] Starting printing (1 copies)
2019-02-14 17:47:54,909 [WARN] Supported printer densities not found
2019-02-14 17:47:55,504 [DEBUG] Paper area: 0,0:89,161
2019-02-14 17:47:55,776 [DEBUG] Paper area: 0,0:89,161
2019-02-14 17:47:56,834 [INFO] Printing complete
```

Of particular note, is in the email chain it was suggested that the client try printing with "rasterize" set to false, which you can see is set here, but they also set "density" to "120" which has a side effect  of forcing pdfs to raster anyways. Assuming the raster is the problem, this needs to be set to "0".

I have some suspicions that the issue might still be with the client's driver configuration though. 
@tresf, did you ever get a response from the client about the raw vs driver printer names? The chain forwarded to me ends there.

**Comment by tresf** [comment]
> I have some suspicions that the issue might still be with the client's driver configuration though.
@tresf, did you ever get a response from the client about the raw vs driver printer names

Yes, I was able to confirm that the raw printer was configured under a different name.

> `density: 120`

Good catch.  This is actually `304.8` dpi because the units are `cm`, which I never caught previously.

Interestingly I've removed and re-added the EPL2 driver (Zebra LP2844 203-dpi printer) and everything's printing fine.  I'll double check with the client.
